packages/cosmic-swingset/src/chain-main.js

```diff
@@ -262,20 +262,26 @@ export default async function main(progname, args, { env, homedir, agcc }) {
         val => stringify(exportMailbox(val)),
       ),
     );
-    const actionQueueRawStorage = makeBufferedStorage(
-      makePrefixedBridgeStorage(
-        sendToChain,
-        `${STORAGE_PATH.ACTION_QUEUE}.`,
-        'setWithoutNotify',
-        x => x,
-        x => x,
-      ),
+    const makeQueueStorage = queuePath => {
+      const queueRawStorage = makeBufferedStorage(
+        makePrefixedBridgeStorage(
+          sendToChain,
+          `${queuePath}.`,
+          'setWithoutNotify',
+          x => x,
+          x => x,
+        ),
+      );
+      return harden({
+        ...queueRawStorage.kvStore,
+        commit: queueRawStorage.commit,
+        abort: queueRawStorage.abort,
+      });
```

Comment:
I think there's a slightly more conventional way to express this.
```suggestion
      const { kvStore, commit, abort } = makeBufferedStorage(
        makePrefixedBridgeStorage(
          sendToChain,
          `${queuePath}.`,
          'setWithoutNotify',
          x => x,
          x => x,
        ),
      );
      return harden({ ...kvStore, commit, abort });
```

Reply:
Duh